File: MiniDOMOffice/Office/LuoghiVisitati/LuoghiVisitatiCursor.vb
Imports minidom.Databases
Imports minidom.Sistema


Partial Class Office

    ''' <summary>
    ''' Cursore sulla tabella dei luoghi attraversati dall'utente per le commissioni
    ''' </summary>
    ''' <remarks></remarks>
    Public Class LuoghiVisitatiCursor
        Inherits DBObjectCursor(Of LuogoVisitato)

        Private m_IDOperatore As New CCursorField(Of Integer)("IDOperatore")
        Private m_IDUscita As New CCursorField(Of Integer)("IDUscita")
        Private m_IndirizzoEtichetta As New CCursorFieldObj(Of String)("Indirizzo_Etichetta")
        Private m_IndirizzoVia As New CCursorFieldObj(Of String)("Indirizzo_Via")
        Private m_IndirizzoCitta As New CCursorFieldObj(Of String)("Indirizzo_Citta")
        Private m_IndirizzoProvincia As New CCursorFieldObj(Of String)("Indirizzo_Provincia")
        Private m_IndirizzoCAP As New CCursorFieldObj(Of String)("Indirizzo_CAP")
        Private m_OraArrivo As New CCursorField(Of Date)("OraArrivo")
        Private m_OraPartenza As New CCursorField(Of Date)("OraPartenza")
        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_Lat As New CCursorField(Of Double)("Lat")
        Private m_Lng As New CCursorField(Of Double)("Lng")
        Private m_Alt As New CCursorField(Of Double)("Alt")
        Private m_IDLuogo As New CCursorField(Of Integer)("IDLuogo")
        Private m_IDPersona As New CCursorField(Of Integer)("IDPersona")
        Private m_TipoMateriale As New CCursorFieldObj(Of String)("TipoMateriale")
        Private m_ConsegnatiAMano As New CCursorField(Of Integer)("ConsegnatiAMano")
        Private m_ConsegnatiPostale As New CCursorField(Of Integer)("ConsegnatiPostale")
        Private m_ConsegnatiAuto As New CCursorField(Of Integer)("ConsegnatiAuto")
        Private m_ConsegnatiAltro As New CCursorField(Of Integer)("ConsegnatiAltro")
        Private m_Progressivo As New CCursorField(Of Integer)("Progressivo")

        Public ReadOnly Property Progressivo As CCursorField(Of Integer)
            Get
                Return Me.m_Progressivo
            End Get
        End Property

        Public ReadOnly Property IDOperatore As CCursorField(Of Integer)
            Get
                Return Me.m_IDOperatore
            End Get
        End Property

        Public ReadOnly Property IDUscita As CCursorField(Of Integer)
            Get
                Return Me.m_IDUscita
            End Get
        End Property

        Public ReadOnly Property Indirizzo_Etichetta As CCursorFieldObj(Of String)
            Get
                Return Me.m_IndirizzoEtichetta
            End Get
        End Property

        Public ReadOnly Property Indirizzo_Via As CCursorFieldObj(Of String)
            Get
                Return Me.m_IndirizzoVia
            End Get
        End Property

        Public ReadOnly Property Indirizzo_Citta As CCursorFieldObj(Of String)
            Get
                Return Me.m_IndirizzoCitta
            End Get
        End Property

        Public ReadOnly Property Indirizzo_Provincia As CCursorFieldObj(Of String)
            Get
                Return Me.m_IndirizzoProvincia
            End Get
        End Property

        Public ReadOnly Property Indirizzo_CAP As CCursorFieldObj(Of String)
            Get
                Return Me.m_IndirizzoCAP
            End Get
        End Property

        Public ReadOnly Property OraArrivo As CCursorField(Of Date)
            Get
                Return Me.m_OraArrivo
            End Get
        End Property

        Public ReadOnly Property OraPartenza As CCursorField(Of Date)
            Get
                Return Me.m_OraPartenza
            End Get
        End Property

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property

        Protected Overrides Function GetModule() As CModule
            Return Office.LuoghiVisitati.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeLuoghiV"
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function

        Public ReadOnly Property Lat As CCursorField(Of Double)
            Get
                Return Me.m_Lat
            End Get
        End Property

        Public ReadOnly Property Lng As CCursorField(Of Double)
            Get
                Return Me.m_Lng
            End Get
        End Property

        Public ReadOnly Property Alt As CCursorField(Of Double)
            Get
                Return Me.m_Alt
            End Get
        End Property

        Public ReadOnly Property IDLuogo As CCursorField(Of Integer)
            Get
                Return Me.m_IDLuogo
            End Get
        End Property

        Public ReadOnly Property IDPersona As CCursorField(Of Integer)
            Get
                Return Me.m_IDPersona
            End Get
        End Property

        Public ReadOnly Property TipoMateriale As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoMateriale
            End Get
        End Property

        Public ReadOnly Property ConsegnatiAMano As CCursorField(Of Integer)
            Get
                Return Me.m_ConsegnatiAMano
            End Get
        End Property

        Public ReadOnly Property ConsegnatiPostale As CCursorField(Of Integer)
            Get
                Return Me.m_ConsegnatiPostale
            End Get
        End Property

        Public ReadOnly Property ConsegnatiAuto As CCursorField(Of Integer)
            Get
                Return Me.m_ConsegnatiAuto
            End Get
        End Property

        Public ReadOnly Property ConsegnatiAltro As CCursorField(Of Integer)
            Get
                Return Me.m_ConsegnatiAltro
            End Get
        End Property

    End Class



End Class

File: MiniDOMOffice/Office/Uscite/LuoghiVisitati/LuoghiVisitatiCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;


namespace minidom
{
    public partial class Office
    {

        /// <summary>
        /// Cursore di <see cref="LuogoVisitato"/>
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class LuoghiVisitatiCursor 
            : minidom.Databases.DBObjectCursor<LuogoVisitato>
        {
            private DBCursorField<int> m_IDOperatore = new DBCursorField<int>("IDOperatore");
            private DBCursorField<int> m_IDUscita = new DBCursorField<int>("IDUscita");
            private DBCursorStringField m_IndirizzoEtichetta = new DBCursorStringField("Indirizzo_Etichetta");
            private DBCursorStringField m_IndirizzoVia = new DBCursorStringField("Indirizzo_Via");
            private DBCursorStringField m_IndirizzoCitta = new DBCursorStringField("Indirizzo_Citta");
            private DBCursorStringField m_IndirizzoProvincia = new DBCursorStringField("Indirizzo_Provincia");
            private DBCursorStringField m_IndirizzoCAP = new DBCursorStringField("Indirizzo_CAP");
            private DBCursorField<DateTime> m_OraArrivo = new DBCursorField<DateTime>("OraArrivo");
            private DBCursorField<DateTime> m_OraPartenza = new DBCursorField<DateTime>("OraPartenza");
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<double> m_Lat = new DBCursorField<double>("Lat");
            private DBCursorField<double> m_Lng = new DBCursorField<double>("Lng");
            private DBCursorField<double> m_Alt = new DBCursorField<double>("Alt");
            private DBCursorField<int> m_IDLuogo = new DBCursorField<int>("IDLuogo");
            private DBCursorField<int> m_IDPersona = new DBCursorField<int>("IDPersona");
            private DBCursorStringField m_TipoMateriale = new DBCursorStringField("TipoMateriale");
            private DBCursorField<int> m_ConsegnatiAMano = new DBCursorField<int>("ConsegnatiAMano");
            private DBCursorField<int> m_ConsegnatiPostale = new DBCursorField<int>("ConsegnatiPostale");
            private DBCursorField<int> m_ConsegnatiAuto = new DBCursorField<int>("ConsegnatiAuto");
            private DBCursorField<int> m_ConsegnatiAltro = new DBCursorField<int>("ConsegnatiAltro");
            private DBCursorField<int> m_Progressivo = new DBCursorField<int>("Progressivo");


            /// <summary>
            /// Costruttore
            /// </summary>
            public LuoghiVisitatiCursor()
            {

            }

            /// <summary>
            /// Progressivo
            /// </summary>
            public DBCursorField<int> Progressivo
            {
                get
                {
                    return m_Progressivo;
                }
            }

            /// <summary>
            /// IDOperatore
            /// </summary>
            public DBCursorField<int> IDOperatore
            {
                get
                {
                    return m_IDOperatore;
                }
            }

            /// <summary>
            /// IDUscita
            /// </summary>
            public DBCursorField<int> IDUscita
            {
                get
                {
                    return m_IDUscita;
                }
            }

            /// <summary>
            /// Indirizzo_Etichetta
            /// </summary>
            public DBCursorStringField Indirizzo_Etichetta
            {
                get
                {
                    return m_IndirizzoEtichetta;
                }
            }

            /// <summary>
            /// Indirizzo_Via
            /// </summary>
            public DBCursorStringField Indirizzo_Via
            {
                get
                {
                    return m_IndirizzoVia;
                }
            }

            /// <summary>
            /// Indirizzo_Citta
            /// </summary>
            public DBCursorStringField Indirizzo_Citta
            {
                get
                {
                    return m_IndirizzoCitta;
                }
            }

            /// <summary>
            /// Indirizzo_Provincia
            /// </summary>
            public DBCursorStringField Indirizzo_Provincia
            {
                get
                {
                    return m_IndirizzoProvincia;
                }
            }

            /// <summary>
            /// Indirizzo_CAP
            /// </summary>
            public DBCursorStringField Indirizzo_CAP
            {
                get
                {
                    return m_IndirizzoCAP;
                }
            }

            /// <summary>
            /// OraArrivo
            /// </summary>
            public DBCursorField<DateTime> OraArrivo
            {
                get
                {
                    return m_OraArrivo;
                }
            }

            /// <summary>
            /// OraPartenza
            /// </summary>
            public DBCursorField<DateTime> OraPartenza
            {
                get
                {
                    return m_OraPartenza;
                }
            }

            /// <summary>
            /// Descrizione
            /// </summary>
            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Office.LuoghiVisitati;
            }

            /// <summary>
            /// Discriminator
            /// </summary>
            /// <returns></returns>
            public override string GetTableName()
            {
                return "tbl_OfficeLuoghiV";
            }
             
            /// <summary>
            /// Latitudine
            /// </summary>
            public DBCursorField<double> Lat
            {
                get
                {
                    return m_Lat;
                }
            }

            /// <summary>
            /// Longitudine
            /// </summary>
            public DBCursorField<double> Lng
            {
                get
                {
                    return m_Lng;
                }
            }

            /// <summary>
            /// Altitudine
            /// </summary>
            public DBCursorField<double> Alt
            {
                get
                {
                    return m_Alt;
                }
            }

            /// <summary>
            /// ID del luogo
            /// </summary>
            public DBCursorField<int> IDLuogo
            {
                get
                {
                    return m_IDLuogo;
                }
            }

            /// <summary>
            /// ID della persona
            /// </summary>
            public DBCursorField<int> IDPersona
            {
                get
                {
                    return m_IDPersona;
                }
            }

            /// <summary>
            /// TipoMateriale
            /// </summary>
            public DBCursorStringField TipoMateriale
            {
                get
                {
                    return m_TipoMateriale;
                }
            }

            /// <summary>
            /// ConsegnatiAMano
            /// </summary>
            public DBCursorField<int> ConsegnatiAMano
            {
                get
                {
                    return m_ConsegnatiAMano;
                }
            }

            /// <summary>
            /// ConsegnatiPostale
            /// </summary>
            public DBCursorField<int> ConsegnatiPostale
            {
                get
                {
                    return m_ConsegnatiPostale;
                }
            }

            /// <summary>
            /// ConsegnatiAuto
            /// </summary>
            public DBCursorField<int> ConsegnatiAuto
            {
                get
                {
                    return m_ConsegnatiAuto;
                }
            }

            /// <summary>
            /// ConsegnatiAltro
            /// </summary>
            public DBCursorField<int> ConsegnatiAltro
            {
                get
                {
                    return m_ConsegnatiAltro;
                }
            }
        }
    }
}